compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -239,9 +242,10 @@ public class QueryDesugar extends BLangNodeVisitor {
     private SymbolEnv queryEnv;
     private boolean containsCheckExpr;
     private boolean withinQuery = false;
-    private boolean afterGroupBy = false;
     private boolean withinLambdaOrArrowFunc = false;
+    private boolean withinCollectClause = false;
```

Comment:
Is it possible to move these variables to `AnalyzerData` object?

Reply:
Still QueryDesugar does not follow that pattern.